src/crypto/x509.js

```diff
@@ -189,7 +193,36 @@ export default class X509Chain {
 
     getParameterValueByName(parameterName) {
         let signerCert = this.certChain[0];
-        return signerCert[parameterName]?.typesAndValues[0]?.value?.valueBlock?.value || null;
+
+        return signerCert[parameterName].typesAndValues|| [];
+    }
+
+    getSubjectValue() {
+        const subjectValues = this.getParameterValueByName('subject');
+        const subjectName = subjectValues.find((v) => v.type === OIDs.commonName);
+
+
+        if (subjectName && subjectName.value?.valueBlock?.value) {
+            return subjectName.value.valueBlock.value;
+        }
+
+        return 'Subject name not found';
+    }
+
+    getIssuerValue() {
+        const issuerValues = this.getParameterValueByName('issuer');
+        const commonName = issuerValues.find((v) => v.type === OIDs.commonName);
+        const organization = issuerValues.find((v) => v.type === OIDs.organization);
+
+        if (commonName && commonName.value?.valueBlock?.value) {
+            return commonName.value.valueBlock.value;
+        }
+
+        if (organization && organization.value?.valueBlock?.value) {
+            return organization.value.valueBlock.value;
+        }
+
+        return 'Issuer name not found'; 
```

Comment:
I'd suggest return null on the API level and replace it with user friendly string before rendering it on UI.
This is a library and it can be used by other parties than Ravnur, they may have their own business rules.